### PR TITLE
Added HSL Nodes

### PR DESCRIPTION
**Pull request by hipsterusername**
Added nodes for Hue Saturation and Luminosity

### CODE REVIEW

File: invokeai/app/invocations/image.py [hunk 1]
```diff
@@ -501,3 +502,150 @@ def invoke(self, context: InvocationContext) -> ImageOutput:
...
+        # Adjust the luminosity (value)
+        hsv_image[:, :, 2] = numpy.clip(hsv_image[:, :, 2] * self.luminosity, 0, 255)
+
+        # Convert image back to BGR color space
+        image = cv2.cvtColor(hsv_image, cv2.COLOR_HSV2BGR)
+
+        # Convert back to PIL format and to original color mode
+        pil_image = Image.fromarray(image[:, :, ::-1], 'RGB').convert('RGBA')
+
+        image_dto = context.services.images.create(
```
Comment (lstein) [inline invokeai/app/invocations/image.py:587]: I see these lines of code (`context.services.images.create()`) repeated verbatim in multiple invocations. This should be turned into a method in a base class somewhere.

**Review by lstein — CHANGES_REQUESTED** [review]
See comment.

**Comment by hipsterusername** [comment]
> See comment.

Comment may be accurate, but is more commentary on the broader definition of images as used in these nodes and others. Probably worth talking about a broader update

cc: @psychedelicious

File: invokeai/app/invocations/image.py (hunk 1 above)
Comment (psychedelicious) [reply invokeai/app/invocations/image.py:587]: I'm a bit late here, but that is a method on the images service class

**Review by psychedelicious — APPROVED** [review]
I've fixed a number of issues related to this getting behind `main` and tested the nodes.

**Review by lstein — APPROVED** [review]
LGTM

**Pull request merged**